packages/source-iotsitewise/src/time-series-data/subscribeToTimeSeriesData.ts

```diff
@@ -23,17 +30,83 @@ export const subscribeToTimeSeriesData =
 
     const errors: Record<string, ErrorDetails> = {};
 
+    let annotations: Annotations = {};
+
+    const alarms: AlarmsStore = {};
+
+    queries.forEach(({ assets }) =>
+      assets.forEach(({ assetId, properties }) =>
+        properties.forEach(({ alarms }) => {
+          if (alarms) {
+            // TODO: get all the alarms where inputProperty = correspondingQuery.propertyId
+            /**
+             * iterate over all compositeModels that are type AWS/ALARM and get the id of AWS/ALARM_STATE property
+             * describe the alarm using alarmModule and find if the inputPropertyId = propertyId
+             *
+             * override queries with queries with alarm streams
+             */
+          }
+        })
+      )
+    );
+
     const emit = () => {
       callback({
-        dataStreams: completeDataStreams({ dataStreams, assetModels }),
+        dataStreams: completeDataStreams({ dataStreams, assetModels, alarms }),
         viewport,
+        annotations,
       });
     };
 
     const { update, unsubscribe } = dataModule.subscribeToDataStreams({ queries, request }, (data) => {
       dataStreams = data.dataStreams;
       viewport = data.viewport;
       emit();
+
+      annotations.y = [];
+      Object.keys(assetModels).forEach(assetId => {
```

Comment:
My understanding of this code is it's providing the correct annotations - though it appears to be doing this every time there is new data (which will be often). Is there something within here that makes it needed to be ran everytime there is new data?

Reply:
You are right, originally I was relying on the asset model being already fetched but now we do that in the alarms module.